src/utils/generateElectricFieldVectorGroupYPoints.js

```diff
@@ -0,0 +1,86 @@
+// a 'vector group' consists of three arrows pointing at a peak/trough of the sine curve (radiation line)
+// every group's middle arrow is centered on a peak/trough
+// given a group number and the current intervalCount, this function determines:
+// (1) the y-position of that middle arrow in a vector group,
+// (2) the direction the vector group is pointing in (1 = right, -1 = left)
+// for example: the first vector group points at the sine curve's first peak, i.e. at the peak after one interval of the sine curve
+// (noting that an 'interval' is one-quarter of the sine curve's period)
+// since the sine curve starts off-screen, this first peak occurs at stageHeight (bottom of canvas) + one curveInterval
+// the next group points at the trough following this peak, occuring two intervals afterwards (peak to 0 = one interval, 0 to trough = one interval)
+// (and so on for subsequent peaks/troughs)
+// finally, we subtract intervalCount * yShiftPerInterval from the group's y position, so that it moves upward along with the sine curve every interval
+const determineVectorGroupYPosition = (
+  groupNumber,
+  intervalCount,
+  yShiftPerInterval,
+  stageHeight,
+  curveInterval,
+) => {
+  return {
+    y:
+      stageHeight +
+      curveInterval +
+      2 * (groupNumber - 1) * curveInterval -
```

Comment:
I guess you can simplify `curveInterval + 2 * (groupNumber - 1) * curveInterval` with 
```(2 * groupNumber - 1) * curveInterval```

Reply:
definitely... when I lay these formulas out this way, it's just to make a bit clearer where they come from. in this case, I'm saying that the first peak occurs at `stageHeight + curveInterval`, so it's helpful to have those two variables at the beginning of the formula.